core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -811,7 +811,16 @@ abstract class Bundle(implicit compileOptions: CompileOptions) extends Record {
       }
     }
 
+    val isAnonFunc = ".*\\$\\$anonfun\\$\\d+$".r
+    def getNonFuncClass(clz: Class[_]): Option[Class[_]] = {
```

Comment:
Can you add the rationale as a comments here, specifically including that this is a workaround for an older version of Scala? The explanation in the issue description should be fine.